fuels-signers/src/wallet.rs

```diff
@@ -223,20 +216,16 @@ impl Signer for Wallet {
         &self,
         message: S,
     ) -> Result<Signature, Self::Error> {
-        let message = message.as_ref();
-        let message_hash = Hasher::hash(message);
-
-        let sig =
-            secp256k1_sign_compact_recoverable(self.private_key.as_ref(), &*message_hash).unwrap();
-        Ok(Signature { compact: sig })
+        let message = Message::new(message);
+        let sig = Signature::sign(&self.private_key, &message);
```

Comment:
if the Signature::sign cannot fail, then we shouldn't return a `Result`. Otherwise, I think you could use `?`

Reply:
`Wallet::sign_message` is provided by the [`Signer`](https://github.com/FuelLabs/fuels-rs/blob/master/fuels-signers/src/lib.rs#L20) trait implementation. As the `Signer` doc mentions, there could be other signing backends in the future, which might want to return a `Result`. And yes, current `Signature::sign` method can't fail.